File: objects/src/main/kotlin/actionsData/MoveAction.kt
package actionsData

import enums.ActionTypeEnum
import kotlinx.serialization.EncodeDefault
import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.Serializable
import objectsData.Loc

@OptIn(ExperimentalSerializationApi::class)
@Serializable
data class MoveAction(
    override val activatorId: Int,
    val newLoc: Loc,
    @EncodeDefault(EncodeDefault.Mode.ALWAYS) override val type: ActionTypeEnum = ActionTypeEnum.MOVE
) : Action()

File: runner/src/main/resources/apis/ts/src/classes/actions/MoveAction.ts
import Action from "./Action";
import Loc from "../Loc";

export default class MoveAction extends Action {
    newLoc: Loc

    constructor(activatorId: number, newLoc: Loc) {
        super("MOVE", activatorId);
        this.newLoc = newLoc;
    }
}